### DISCUSSION TITLE
Multilingual on field or collection level

### DISCUSSION THREAD

**Discussion by khakimvinh** [pocketbase/pocketbase#154] [Q&A] [answered]
Hi everyone, 

Is there any way to handle multilingual on field or collection level? 
I can't find anything related to language in official docs or the discussion threads hence created this one. I think it's not currently supported, but would love to know any suggestions on the matter. 

What I think so far:
Field level multilingual not really possible. 
Collection level: The simplest way I can think of would be create langCode field in the collection. But the I can't use the default "id" to be the item id, since different languages of same item will have different id.

**Comment by ganigeorgiev** [comment, marked as answer]
Hi, currently there is no built-in support for managing data in different languages and it is left to the developers to handle this themselves by:

- creating multiple fields for each language in the same collection (eg. `name_en`, `name_de`, etc.)
- creating multiple collections for each language (eg. `posts_en`, `posts_bg`, etc.)

Another approach (but complicates a little managing the data) could be also to combine them both:
1. Create multiple collections to store the content for each language you want to support, eg. `posts_content_en`, `posts_content_bg`, etc.
2. Create you main entity collection, eg. `posts` and **add relation fields** pointing to the collections created in step 1 (eg. `en`, `bg`, etc.).

By doing this, you will be able to search for the multilingual content through the main collection `posts` like `en.name = 'test'` and in addition you can make use of the `expand` query parameter and let the API autoload your multilingual content:
```js
const record = await client.Records.getOne("posts", "YOUR_RECORD_ID", { expand: "en,bg" });

// could result in:
{
  "@collectionId": "a98f514eb05f454",
  "@collectionName": "posts",
  "id": "ae40239d2bc4477",
  "updated": "2022-06-25 11:03:50.052",
  "created": "2022-06-25 11:03:35.163",
  "en": "a98f514eb05f454",
  "bg": "eb5145405f4a98f",
  "@expand": {
    "en": {
      "id": "a98f514eb05f454",
      "name": "My en title",
      ...
    },
    "bg": {
      "id": "eb5145405f4a98f",
      "name": "My bg title",
      ...
    }
  }
}
```

**Comment by Lex-2008** [comment]
**UPDATE**: @ganigeorgiev was a bit faster than me when composing his answer, but I'll let my answer stay here for history.

I don't know your usecase, but in my case (few languages well-known in advance), I just create several fields with language code appended. For example, "posts" collection has fields "title_en", "title_no", "text_en", "text_no" (for English and Norwegian). Post editor exposes all these entries, so author has to provide all texts in all languages. Client-side decides what language to present to user (based on cookie, or navigator.languages), and appends it to field name, something like this:
```
title=document.getElementById('title');
title.innerText=record['title_'+lang];
```

Another option might be to make all fields be of "json" type and store in them objects like this:
```
{
  "en": "blog title in English",
  "no": "bloggtittel på norsk"
}
```

You will have to make a special interface to fill this json, of course, instead of asking your users to compose json themselves :) Client side then is just a bit better:
```
title=document.getElementById('title');
title.innerText=record.title[lang];
```

**Comment by khakimvinh** [comment]
Thanks for all the suggestions. I've learned much from this!
